connectors/sources/tests/test_s3.py

```diff
@@ -13,6 +14,12 @@
 from connectors.sources.tests.support import assert_basics, create_source
 
 
+@pytest.fixture(scope="session", autouse=True)
+def execute_before_all_tests():
+    os.environ["AWS_ACCESS_KEY_ID"] = "access_key"
```

Comment:
You need to check first if it's there, so you don't overwrite an existing key:

```
if 'AWS_ACCESS_KEY_ID' in os.environ:
    return
```

Reply:
you should also clean up on tear down any set keys in `os.environ`.

Reply:
Hey Tarek, We already checked this thing to unset the keys, However it is not required because once the the test cases gets completed, access key and secret key is automatically removed from the os environment.
let me know if you have any doubt will share the video for the same